commit f8fa70f392fab8697b6358bb7e4f8c3e8429ec7b
Author: Mark Rutland <mark.rutland@arm.com>
Date:   Tue Jul 12 16:28:01 2016 +0100

    arm64: localise Image objcopy flags
    
    We currently define OBJCOPYFLAGS in the top-level arm64 Makefile, and
    thus these flags will be passed to all uses of objcopy, kernel-wide, for
    which they are not explicitly overridden. The flags we set are intended
    for converting vmlinux (and ELF) into Image (a raw binary), and thus the
    flags chosen are problematic for some other uses which do not expect a
    raw binary result, e.g. the upcoming lkdtm rodata test:
    
      http://www.openwall.com/lists/kernel-hardening/2016/06/08/2
    
    This patch localises the objcopy flags such that they are only used for
    the vmlinux -> Image conversion.
    
    Signed-off-by: Mark Rutland <mark.rutland@arm.com>
    Cc: Will Deacon <will.deacon@arm.com>
    Acked-by: Kees Cook <keescook@chromium.org>
    Tested-by: Laura Abbott <labbott@redhat.com>
    Signed-off-by: Catalin Marinas <catalin.marinas@arm.com>

diff --git a/arch/arm64/boot/Makefile b/arch/arm64/boot/Makefile
index 305c552b5ec1..1f012c506434 100644
--- a/arch/arm64/boot/Makefile
+++ b/arch/arm64/boot/Makefile
@@ -14,6 +14,8 @@
 # Based on the ia64 boot/Makefile.
 #
 
+OBJCOPYFLAGS_Image :=-O binary -R .note -R .note.gnu.build-id -R .comment -S
+
 targets := Image Image.gz
 
 $(obj)/Image: vmlinux FORCE